### PR TITLE
:es6: onSlotRender

### PR DESCRIPTION
**Pull request by regiskuckaertz**
There is some synergy with `onSlotLoad` but I'll normalise later.

### CODE REVIEW

**Review by JonNorman — APPROVED** [review]
I don't want to approve this as it will cause more merge conflicts on my PR, but it looks good 😢 ...:+1:

**Comment by regiskuckaertz** [comment]
I'll wait for yours to go first 😆

**Comment by regiskuckaertz** [comment]
@JonNorman mind having a quick look, after my latest changes?

**Comment by regiskuckaertz** [comment]
Most notably there was a call to an undefined API `getTargetingMap` that I replaced with `getTargetingKeys` in `reportEmptyResponse`

**Review by JonNorman — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/dfp/Advert.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ type Resolver = (x: boolean) => void;
  * One size       - `data-mobile="300,50"`
  * Multiple sizes - `data-mobile="300,50|320,50"`
  */
-const createSizeMapping = (attr: string): Array<AdSize> =>
+const createSizeMapping = (attr: string): Array<SingleSize> =>
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/dfp/Advert.js]: I think `SingleSize` initially seems verbose as a type name. It's both grammatically singular and explicitly qualified as singular. But I can see an issue maybe caused by `GeneralSize`. Maybe we could call
`GeneralSize` -> `AdSize`?

File: tools/es5to6.json [hunk 2]
```diff
@@ -21,9 +21,7 @@
     "projects/commercial/modules/creatives/scrollable-mpu-v2.js",
     "projects/common/modules/atoms/youtube.js"
   ],
-  "Regis Kuckaertz": [
-    "projects/commercial/modules/dfp/on-slot-render.js"
-  ],
+  "Regis Kuckaertz": [],
```
Comment (rich-nguyen) [inline tools/es5to6.json]: Welcome to the club!

**Review by rich-nguyen — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/dfp/Advert.js (hunk 1 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/dfp/Advert.js]: Yes, absolutely. I merely copied the definitions from the GPT reference but they are off putting. I see synergies with this file and the `ad-sizes` module, can we try to reconcile them? Maybe by updating `AdSize` this way:
```
type GuAdSize = SingleSizeArray* & { switchUnitId: ?number, toString: (_: void) => string }
```
NB: one could also leverage Flow support for literal types
```
type MPUSwitchId = 228;
type LeaderboardSwitchId = 229;
type BillboardSwitchId = 229;
type SwitchUnitId = MPUSwitchId | LeaderboardSwitchId | BillboardSwitchId;
```
and update the signature of `getAdSize` accordingly.

What say you?

Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/dfp/Advert.js]: Yes to the literal types, definitely. Seems like a great match.

Not so sure about the look of `GuAdSize`, i'm not sure if we really intend to specify that a parameter of the intersection type above is a useful contract to be used by other modules. Modelling-wise, I think it makes most sense  if `ad-sizes:AdSize` either contains a `SingleSize | MultiSize` field, or the more general types in `types.js` are modified to be usable in all modules (that can be achieved by merging the fields `width`, `height`, `switchUnitId` and `toString` into the general type. Might be a bit big for this PR?

Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/dfp/Advert.js]: Yeah, let's leave it aside for now. (besides I conflated an array type with an object type 😊 )

thanks for your advice!

**Pull request merged**